### ISSUE TITLE
Internal error when using a type with collided name from another project

### ISSUE THREAD

**Issue by auduchinok** [dotnet/fsharp#17828]
This is likely the same cause as in #17827, however, the misleading error reported there should be fixed separately from this one, as this is likely about resolving references during the unpickling of the serialized F# metadata.

Consider this library project:

File1.fs:
```fsharp
namespace Ns.TypeName
```

File2.fs:
```fsharp
namespace Ns

type TypeName = { Field: int }

module Module =
    let f (tn: TypeName) = ()
```

Reference this project from the following console app one:
```Program.fs
module Ns.Another

Module.f null
```

There's an internal error when trying to use the function that uses the type:

<img width="519" alt="Screenshot 2024-10-02 at 10 55 48" src="https://github.com/user-attachments/assets/d95d0316-1481-4b79-9894-d9b1e751e621">

Here's the repro project: [SameNameNamespaceType.zip](https://github.com/user-attachments/files/17221650/SameNameNamespaceType.zip).

**Comment by edgarfgp** [comment]
Similar to https://github.com/dotnet/fsharp/issues/16970

**Comment by auduchinok** [comment]
> Similar to https://github.com/dotnet/fsharp/issues/16970

Yes, indeed, thanks! However, these might still be separate issues in the code.
